src/components/DownloadButton.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { set } from 'idb-keyval';
+import GetAppIcon from '@material-ui/icons/GetApp';
+import { videoOrArticle, VideoData, ArticleData } from '../store/resourcesSlice'
+import { videoStore } from '../index'
+import Button from '@material-ui/core/Button';
+import { useDispatch } from 'react-redux';
+import { setResourceIsCached } from '../store/resourcesSlice';
+
+type DownloadButtonProps = {
+  id: number,
+  videoData?: VideoData | ArticleData,
+}
+
+const DownloadButton = ({ id, videoData }: DownloadButtonProps) => {
+  const dispatch = useDispatch();
+
+  async function downloadVideo(resId, videoStore, videoData) {
+    let downloadUrl = videoOrArticle(videoData) ? videoData.downloadUrl : null
```

Comment:
Just curious — since we aren't doing any actual downloading unless the resource is a video, why are we displaying a download button either way (even if it's an article)?

Reply:
Ah, good catch, it should only show up for article -- I'll switch the input to only take in VideoData!